### ISSUE TITLE
How to locally host Connect Installers in React environment

### ISSUE THREAD

**Issue by LKJREAL** [IBM/aspera-connect-sdk-js#34]
Dear dwosk,
I'm clear that how to host ConnectSDK locally in Normal(vanilla) Javascript.
But can't code how to host Connect Installers locally in React.
.......
This is my example which didn't work.
.......

import './App.css';
import { Connect, ConnectInstaller } from '@ibm-aspera/connect-sdk-js';

const connect = new Connect();
const options = { sdkLocation: "//localhost:3000/connectsdk/v4" };
// const options = {};
const connectInstaller = new ConnectInstaller(options);
connect.initSession();

  
  function statusEventListener(eventType, data) {
    if (eventType === Connect.EVENT.STATUS && data === Connect.STATUS.INITIALIZING) {
      connectInstaller.showLaunching();
    } else if (eventType === Connect.EVENT.STATUS && data === Connect.STATUS.EXTENSION_INSTALL) {
      connectInstaller.showExtensionInstall();
    } else if (eventType === Connect.EVENT.STATUS && data === Connect.STATUS.FAILED) {
       connectInstaller.showDownload();
    } else if (eventType === Connect.EVENT.STATUS && data === Connect.STATUS.OUTDATED) {
       connectInstaller.showUpdate();
    } else if (eventType === Connect.EVENT.STATUS && data === Connect.STATUS.RUNNING) {
       connectInstaller.connected();
    }
 };

function transferEventListener(type, allTransfersInfo) {
  if (type === Connect.EVENT.TRANSFER) {
    console.log('Received transfer event!')
  }
};

  connect.addEventListener(Connect.EVENT.STATUS, statusEventListener );
  connect.addEventListener(Connect.EVENT.TRANSFER, transferEventListener);

  connect.initSession();

  function testAspera() {

  const transferSpec = {
  paths: [
  {
  source: 'aspera-test-dir-large/100MB'
  }
  ],
  remote_host: 'demo.asperasoft.com',
  remote_user: 'aspera',
  remote_password: 'demoaspera',
  direction: 'receive'
  };

  var connectSpec = {
    "allow_dialogs" : true,
    "return_paths" : false,
    "return_files" : true,
    "use_absolute_destination_path" : false
  };

  var transferCallback = {
    success: function(result) {
      console.log('Start Aspera Transfer');
    },
    error: function(result) {
      console.log('Start Failed by Console message');
    }
  };

  connect.startTransferPromise(transferSpec, connectSpec, transferCallback);
}

function App() {

return (
<div className="App">

<input type="button" value={'Transfer Test'} onClick={testAspera}/>

</div>
);
}

export default App;

**Comment by LKJREAL** [comment]
If Connect extension was installed already, it shows no problem.
If Connect extension is not installed, it shows "IBM Aspera Connect Starting..." and don't display "Extension install UI".

**Comment by LKJREAL** [comment]
If I update and save sdkLocation at any point when webserver is on, chrome browse shows like this.
-----------------------------------------------------------------
 
![image](https://user-images.githubusercontent.com/111216349/194698243-ea83f9f2-e9d9-4dc7-a0db-9471a985581c.png)

**Comment by LKJREAL** [comment]
The below step works fine.
.....
const options = { sdkLocation: '//192.168.0.12:3000/abc' }; //
const connectInstaller = new ConnectInstaller(options);
connect.initSession();
.....
I put versions.js in //192.168.0.12:3000/abc
and
I put ConnectInstallers_4.2.3.executables in //192.168.0.12:3000/abc/bin
.....

**Comment by LKJREAL** [comment]
Is there official IBM Aspera guide for local hosting in React with Connect SDK 5.0.0 ? 
Especially Global Companies want to locally host SDK because they use in closed network(no access to internet).
Best regards.
KJ.

**Comment by LKJREAL** [comment]
moved this question to "Discussions"

**Comment by dwosk** [comment]
It looks like there is a syntax error in your `versions.js` file. Where did you get this installer bundle from?

We've actually just pushed a docker image that can be used to host the Connect installers for this purpose, although we haven't updated the documentation yet.

Try the following:
1. `docker pull icr.io/ibmaspera/connect-deployer:4.2.3`
2. Serve installers over HTTP: `docker run --rm -it -p 80:80 icr.io/ibmaspera/connect-deployer:4.2.3`
3. OR serve installers over HTTPS: `docker run --rm -it -v /path/to/your/ssl/certs:/etc/ssl/certs 443:443 icr.io/ibmaspera/connect-deployer:4.2.3`

After (2) or (3), the Connect installers will be available (for example) at: `http://127.0.0.1/connect`.

A simple test:
```
curl -i -X GET http://127.0.0.1/connect/references.json

HTTP/1.1 200 OK
Server: nginx/1.23.1
Date: Mon, 19 Sep 2022 20:43:10 GMT
Content-Type: application/json
Content-Length: 2741
Last-Modified: Mon, 19 Sep 2022 20:00:53 GMT
Connection: keep-alive
ETag: "6328ca75-ab5"
Accept-Ranges: bytes

{"title":"Aspera Connect Desktop Installers","updated":"2022-09-19T20:00:53.365Z","entries":[{"title":"Aspera Connect for Windows 64-bit","platform":{"os":"win64"},"version":"4.2.2.135","updated":"2022-09-19T20:00:53.365Z","links":[{"title":"Windows 64-bit Installer","type":"application/octet-stream","href":"bin/ibm-aspera-connect_4.2.2.135_win64.msi","hreflang":"en","rel":"enclosure"},{"title":"Windows 64-bit Installer","type":"application/octet-stream","href":"bin/ibm-aspera-connect_4.2.2.135_win64.msi","hreflang":"en","rel":"enclosure-one-click"},{"title":"Aspera Connect HTML Documentation for Windows","type":"text/html","href":"https://www.ibm.com/docs/en/aspera-connect/4.2?topic=aspera-connect-user-guide-windows","hreflang":"en","rel":"documentation"},{"title":"Aspera Connect Release Notes for Windows","type":"text/html","href":"https://www.ibm.com/docs/en/aspera-connect/4.2?topic=release-notes-aspera-connect-42","hreflang":"en","rel":"release-notes"}]},{"title":"Aspera Connect for macOS","platform":{"os":"macOS"},"version":"4.2.2.135","updated":"2022-09-19T20:00:53.365Z","links":[{"title":"macOS Installer","type":"application/octet-stream","href":"bin/ibm-aspera-connect_4.2.2.135_macOS_x86_64.pkg","hreflang":"en","arch":"x86_64","rel":"enclosure"},{"title":"macOS Installer","type":"application/octet-stream","href":"bin/ibm-aspera-connect_4.2.2.135_macOS_x86_64.pkg","hreflang":"en","arch":"x86_64","rel":"enclosure-one-click"},{"title":"Aspera Connect for macOS HTML Documentation","type":"text/html","href":"https://www.ibm.com/docs/en/aspera-connect/4.2?topic=aspera-connect-user-guide-macos","hreflang":"en","rel":"documentation"},{"title":"Aspera Connect for macOS Release Notes","type":"text/html","href":"https://www.ibm.com/docs/en/aspera-connect/4.2?topic=release-notes-aspera-connect-42","hreflang":"en","rel":"release-notes"}]},{"title":"Aspera Connect for Linux 64-bit","platform":{"os":"linux"},"version":"4.2.2.135","updated":"2022-09-19T20:00:53.365Z","links":[{"title":"Linux 64-bit Installer","type":"application/octet-stream","href":"bin/ibm-aspera-connect_4.2.2.135_linux.tar.gz","hreflang":"en","rel":"enclosure"},{"title":"Linux 64-bit Installer","type":"application/octet-stream","href":"bin/ibm-aspera-connect_4.2.2.135_linux.tar.gz","hreflang":"en","rel":"enclosure-one-click"},{"title":"Aspera Connect for Linux HTML Documentation","type":"text/html","href":"https://www.ibm.com/docs/en/aspera-connect/4.2?topic=aspera-connect-user-guide-linux","hreflang":"en","rel":"documentation"},{"title":"Aspera Connect for Linux Release Notes","type":"text/html","href":"https://www.ibm.com/docs/en/aspera-connect/4.2?topic=release-notes-aspera-connect-42","hreflang":"en","rel":"release-notes"}]}]}%  
```

**Comment by LKJREAL** [comment]
@dwosk 
Wow great news! It works fine!
I can't thank you enough. Your support was highly appreciated!
Sincerely.
LKJREAL.
